src/Contrib/Jaeger/SpanConverter.php

```diff
@@ -0,0 +1,190 @@
+        return (string) $value;
+    }
+
+    private function buildTags(array $tagPairs): array
+    {
+        $tags = [];
+        foreach ($tagPairs as $key => $value) {
+            $tags[] = $this->buildTag($key, $value);
+        }
+
+        return $tags;
+    }
+
+    private function buildTag(string $key, string $value): Tag
+    {
+        return new Tag([
+            'key' => $key,
+            'vType' => TagType::STRING,
+            'vStr' => $value,
+        ]);
+
+        //TODO - determine if any of this additional handling is still needed, or if it's all dead-code (i.e. $value must be a string at this point always)
+
+        // if (is_bool($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::BOOL,
+        //         'vBool' => $value,
+        //     ]);
+        // } elseif (is_string($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::STRING,
+        //         'vStr' => $value,
+        //     ]);
+        // } elseif (null === $value) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::STRING,
+        //         'vStr' => '',
+        //     ]);
+        // } elseif (is_integer($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::LONG,
+        //         'vLong' => $value,
+        //     ]);
+        // } elseif (is_numeric($value)) {
+        //     return new Tag([
+        //         'key' => $key,
+        //         'vType' => TagType::DOUBLE,
+        //         'vDouble' => $value,
+        //     ]);
+        // }
+
+        // error_log('Cannot build tag for ' . $key . ' of type ' . gettype($value));
+
+        // throw new \Exception('unsupported tag type');
+    }
+
+    private static function convertOtlpToJaegerIds(SpanDataInterface $span): array
+    {
+        if (PHP_INT_SIZE < 8) {
+            $humanReadableIntSize = PHP_INT_SIZE*8;
+
+            throw new RuntimeException("Integrating with Jaeger requires usage of 64 bit integers, but your current platform is $humanReadableIntSize bit. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#ids for more information.");
+        }
+
+        [
+            'traceIdLow' => $traceIdLow,
+            'traceIdHigh' => $traceIdHigh
+        ] = self::convertOtlpToJaegerTraceIds($span->getContext()->getTraceID());
+
+        $spanId = intval($span->getContext()->getSpanID(), 16);
+        $parentSpanId = intval($span->getParentSpanId(), 16);
+
+        return [
+            'traceIdLow' => $traceIdLow,
+            'traceIdHigh' => $traceIdHigh,
+            'spanId' => $spanId,
+            'parentSpanId' => $parentSpanId,
+        ];
+    }
+
+    private static function convertOtlpToJaegerTraceIds(string $traceId): array
```

Comment:
Is there any benefit to this method being static?

Reply:
I'm not sure. I don't really know what the idiomatic way of handling small side-effect-free helper functions like this is in PHP, but whatever that is I'd like to learn it (just been using static as the closest thing I'm aware of)

Overall I don't have a strong preference any way, so I can update it however folks think is best

Reply:
@Grunet 
I use static methods for functions like that as well, if they do not touch any property. 
In that way it's clear that it's a helper function which could potentially be extracted.